entities/Entity.h

```diff
@@ -48,7 +46,7 @@ class Entity {
 
     void setId(const EntityId &id);
 
-    const std::unique_ptr<Geometry> &getGeometry() const;
+    virtual const Geometry *getGeometry() const = 0;
```

Comment:
Better to use shared_ptr then raw pointer.

Reply:
Fixed